crates/macro/src/html_tree/html_component.rs

```diff
@@ -392,9 +395,13 @@ struct ListProps {
 impl Parse for ListProps {
     fn parse(input: ParseStream) -> ParseResult<Self> {
         let mut props: Vec<HtmlProp> = Vec::new();
+
         while HtmlProp::peek(input.cursor()).is_some() {
             props.push(input.parse::<HtmlProp>()?);
         }
+        if input.cursor().token_stream().to_string().contains("with") {
```

Comment:
This could catch errors not related to the `with` syntax like so:
```rust
error: Using special syntax `with props` along with named prop is not allowed
  --> $DIR/html-component-fail.rs:63:26
   |
63 |     html! { <Child int=1 "with" /> };
   |                          ^^^^^^
```

I'd prefer this approach:
```rust
if let Some(ident) = input.cursor().ident() {
  if ident.0 == "with" {
    // ... 
  }
}
```